puppeteer-screenshot/screenshot.js

```diff
@@ -4,24 +4,21 @@ const { getInt, getUrlFromPath, isValidUrl } = require('./validator');
 
 module.exports = async function (req, res) {
     try {
-        const { pathname = '/', query = {} } = parse(req.url, true);
-        const { type = 'png', quality, fullPage } = query;
+        const { pathname = '/' } = parse(req.url);
+        const { type = 'png', quality, fullPage } = req.query;
         const url = getUrlFromPath(pathname);
         const qual = getInt(quality);
         if (!isValidUrl(url)) {
-            res.statusCode = 400;
             res.setHeader('Content-Type', 'text/html');
-            res.end(`<h1>Bad Request</h1><p>The url <em>${url}</em> is not valid.</p>`);
+            res.status(400).send(`<h1>Bad Request</h1><p>The url <em>${url}</em> is not valid.</p>`);
         } else {
             const file = await getScreenshot(url, type, qual, fullPage);
-            res.statusCode = 200;
             res.setHeader('Content-Type', `image/${type}`);
-            res.end(file);
+            res.status(200).send(file);
         }
     } catch (e) {
-        res.statusCode = 500;
-        res.setHeader('Content-Type', 'text/html');
-        res.end('<h1>Server Error</h1><p>Sorry, there was a problem</p>');
         console.error(e.message);
+        res.setHeader('Content-Type', 'text/html');
+        res.status(500).send('<h1>Server Error</h1><p>Sorry, there was a problem</p>');
```

Comment:
```suggestion
        res.status(500).send('<h1>Unexpected Error</h1><p>Sorry, there was a problem</p>');
```

We should avoid using the word "server" since we are a "serverless" provider 😉